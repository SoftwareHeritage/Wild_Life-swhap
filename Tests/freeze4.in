freeze(p(A)), A=1?

freeze(p(A)), A=1, fail?
freeze(p(A)), A={1;2;3}, fail?
freeze(p(A))?
A=1?
A={1;2;3},fail?

freeze(var(A))?
freeze(A<B)?
freeze(write(X))?
w(A:int) :- write(A).
freeze(w(A))?
A=234?


