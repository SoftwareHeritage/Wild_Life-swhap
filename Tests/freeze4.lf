
p(X:int) :- write(X), nl.

