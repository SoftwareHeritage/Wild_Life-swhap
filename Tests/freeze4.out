> 
*** Yes
> > 1

*** Yes
A = 1.
--1> 
*** No
> 1

*** No
> 1
2
3

*** No
> 
*** Yes
A = @~.
--1> 1

*** Yes
A = 1.
----2> 
*** No
A = 1.
----2> 
*** No
A = @~.
--1> 
*** No
A = @~.
--1> 
*** No
A = @~.
--1> 
*** No
A = @~.
--1> 
*** Yes
A = @~.
--1> 
*** Yes
A = @~~.
----2> 234234

*** Yes
A = 234.
------3> 
*** No
A = @~~.
----2> 
*** No
A = @~.
--1> 
